### PR TITLE
test(server): added tests for graphql server

### PR DESCRIPTION
**Pull request by osher-sade**

### CODE REVIEW

File: src/server/graphql-server.ts [hunk 1]
```diff
@@ -57,12 +57,12 @@ export class PolarisGraphQLServer implements GraphQLServer {
                         body: context.request.body,
                     };
                 } catch (e) {
-                    this.polarisLogger.error('headers error', { throwable: e });
+                    this.polarisLogger.error('Headers error', { throwable: e });
```
Comment (yarinvak) [inline src/server/graphql-server.ts]: write a more descriptive message
and what about tests for getHeaders() method?
does it fails if someone sends an header that is not listed in the headers schema (bad)? 
does it fails if someone does not send an header that is listed in the headers schema (bad)?

File: src/middlewares/polaris-middleware.ts [hunk 2]
```diff
@@ -54,8 +55,10 @@ export class PolarisMiddleware implements Middleware {
     shouldBeReturned(params: ResponseMiddlewareParams, headersConfig: HeadersConfiguration) {
         return (
             !(params.root && isRepositoryEntity(params.root)) ||
-            ((headersConfig.realityId === false || RealityIdFilter.shouldBeReturned(params)) &&
-                (headersConfig.dataVersion === false || DataVersionFilter.shouldBeReturned(params)))
+            (SoftDeleteFilter.shouldBeReturned(params) &&
+                ((headersConfig.realityId === false || RealityIdFilter.shouldBeReturned(params)) &&
```
Comment (yarinvak) [inline src/middlewares/polaris-middleware.ts]: headersConfig -> middlewaresConfig

Comment (yarinvak) [inline src/middlewares/polaris-middleware.ts]: realityId -> allowRealityMiddleware

File: src/middlewares/polaris-middleware.ts [hunk 3]
```diff
@@ -54,8 +55,10 @@ export class PolarisMiddleware implements Middleware {
...
+                    (headersConfig.dataVersion === false ||
```
Comment (yarinvak) [inline src/middlewares/polaris-middleware.ts]: dataVersion -> allowDataVersionMiddleware

File: src/middlewares/polaris-middleware.ts (hunk 2 above)
Comment (yarinvak) [inline src/middlewares/polaris-middleware.ts]: x===false -> !x

File: src/server/graphql-server.ts (hunk 1 above)
Comment (osher-sade) [reply src/server/graphql-server.ts]: the descriptive message will appear in the throwable.
getHeaders method will be tested separately.

**Pull request merged**